pkg/bgpv1/agent/controller.go

```diff
@@ -431,62 +429,51 @@ func (c *Controller) FullWithdrawal(ctx context.Context) {
 }
 
 // applyPolicyDefaults applies default values on the CiliumBGPPeeringPolicy.
+// This defaulting is normally done by kube-apiserver (based on CRD default markers), but ensured here for non-k8s environments, tests, etc.
```

Comment:
+1 to use `Int32Deref`, this method with a bunch of `if ( ptr == nil || *ptr ==0 ) {...}` will grow as more optional fields are added.

Reply:
Yeah, I think you understood correctly @YutaroHayakawa . We can do one of the following:

1. always deference with care, e.g. via `pointer.Int32Deref` and we could remove `applyPolicyDefaults()`
2. fully rely on `applyPolicyDefaults()` and always deference directly like `*this`
3. do both of the above, just in case

At the moment we are at the option 3) which I guess is the least ideal?

Reply:
We discussed this a bit more with @YutaroHayakawa and decided to allow direct pointer dereferences. Apart from making it easier for reconciler developers, it is also safer if we always default before passing the API structs to the reconcilers (in case some developer would forget a nil check / safe dereference). In https://github.com/cilium/cilium/pull/26040/commits/06f1756df3d76161aecb76cc203e0e98d15e9834 I cleaned this up and moved the defaulting logic to the API package, so that it can be used from other packages as well (was needed for unit tests).